### ISSUE TITLE
PSVITA Does not automatically flush save data when saving 

### ISSUE THREAD

**Issue by Zadeis** [libretro/gpsp#221]
So I am making a issue as referenced and recommended in the issue [here](https://github.com/libretro/RetroArch/issues/10643). This core either does not support the libretro api for saving or is not automatically flush the save data to the save file.

With the PS Vita being a handheld having to specifically save then close the core before swiping away and closing retroarch from the home screen is inconvenient.

What information can I provide to facilitate this problem being remedied? I looked at the current open and closed issues and didn't find anything similar so my apologies if this has been brought up already.

**Comment by andymcca** [comment]
Hi!  You can specify gpsp or libretro as the save method in the Core Options.  Does it not work as you describe if you select libretro?

**Comment by Zadeis** [comment]
I seem to be overlooking the option you are referring to. I've checked the options/information under Settings > Core and nothing seems to resemble what you are describing. 

However upon starting a game there is an option for "Backup Save Method" which seems to have remedied what I am experiencing. Unfortunately now I seem to have another problem which I can't seem to find how to set it as the default behavior for the core. I'll look into that further.

Which I guess brings me to the question of why this isn't the default behavior? I would assume many would assume that upon saying they wouldn't have to specifically close the running content to flush to the save file.

**Comment by andymcca** [comment]
On the version of RetroArch I'm using (1.13.0), once I set that option whilst in-game and via Quick Menu -> Core Options, and then restart, it stays as the default for future games.

David tells me that the libretro save method is somewhat broken, hence why it's not currently default.  I've been working with him this week to try and improve my PR for save methods (see #220). This is for a few games which don't detect type correctly (i.e. Dragon Ball games, Yggdra Union etc).  So I'm continuing to test this as and when I get time to work towards a good solution for the gpsp saves - by extension it will probably improve the libretro ones, but there are apparently issues with the Libretro API which need to be resolved as well.

**Comment by Zadeis** [comment]
Ok so it sounds like progress is being made then and what you described with the setting applying to all games seems to be the case. When I first set it maybe I did something to set it back to make me think it wasn't working for all.

I appreciate the information and hopefully if others encounter this they can be lead to a solution until the behavior we discussed becomes default.

My apologies for using the issue tracker as a bit of a forum post in the end since I know that isn't the proper use case. I'll probably make a donation in the near future due to this positive interaction

Thanks again!
